viewing_party/party.py

```diff
@@ -1,23 +1,181 @@
-    pass
+    movie_info = {}
+    if not title or not genre or not rating:
+        return None
+
+    # if title and genre and rating:
+
+    movie_info["title"] = title
+    movie_info["genre"] = genre
+    movie_info["rating"] = rating
+
+    return movie_info
+
+result = create_movie("", "Horror", "5")
+print(result)
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+
+    return user_data
+
 
-# -----------------------------------------
 # ------------- WAVE 2 --------------------
-# -----------------------------------------
+def get_watched_avg_rating(user_data):
+    ratings_list = []
+    if not user_data["watched"]:
+        return 0.0
+    for movie in user_data["watched"]:
+        ratings_list.append(movie["rating"])
+
+    average_rating = sum(ratings_list) / len(ratings_list)
+    return average_rating
 
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    most_watched_genre = ""
+    if not user_data["watched"]:
+        return None
+
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_count:
+            genre_count[movie["genre"]] = 1
+        else:
+            genre_count[movie["genre"]] += 1
+
+    most_watched_genre = max(genre_count, key=genre_count.get)
+    return most_watched_genre
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+def get_unique_watched(user_data):
+    user_watched = []
+    for movie in user_data["watched"]:
+        user_watched.append(movie["title"])
+
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in friends_watched:
+                friends_watched.append(movie["title"])
+
+    user_watched_set = set(user_watched)
+    friends_watched_set = set(friends_watched)
+
+    unique_movies_watched_user_only = user_watched_set - friends_watched_set
+
+    unique_movies_watched_user_only_list = []
+    for movie in user_data["watched"]:
+        if movie["title"] in unique_movies_watched_user_only:
+            unique_movies_watched_user_only_list.append(movie)
+
+    return unique_movies_watched_user_only_list
+
+def get_friends_unique_watched(user_data):
+    user_watched = []
+    for movie in user_data["watched"]:
+        user_watched.append(movie["title"])
+
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in friends_watched:
+                friends_watched.append(movie["title"])

```

Comment:
Consider how similar these two loops are to the loops in the previous function. When refactoring, be on the lookout for similar code like this. Feel free to write your own helper function that could be reused to turn these lists into sets!